apps/emqx_management/src/emqx_mgmt_api_subscriptions.erl

```diff
@@ -310,7 +310,12 @@ consume_n_matching(Map, Pred, N, S) ->
 consume_n_matching(_Map, _Pred, _N, [], Acc) ->
     {lists:reverse(Acc), []};
 consume_n_matching(_Map, _Pred, 0, S, Acc) ->
-    {lists:reverse(Acc), S};
+    case emqx_utils_stream:next(S) of
+        [] ->
+            {lists:reverse(Acc), []};
```

Comment:
```suggestion
            {lists:reverse(Acc), emqx_utils_stream:empty()};
```
?

Reply:
Actually, that's part of the fix: there is this check to determine if `hasnext = true`: https://github.com/thalesmg/emqx/blob/0dee2d67b1645b2ae2bcb46d2817c56b03b89f6f/apps/emqx_management/src/emqx_mgmt_api_subscriptions.erl#L258